lib/ruby_indexer/lib/ruby_indexer/index.rb

```diff
@@ -469,16 +469,23 @@ def instance_variable_completion_candidates(name, owner_name)
 
     # Synchronizes a change made to the given indexable path. This method will ensure that new declarations are indexed,
     # removed declarations removed and that the ancestor linearization cache is cleared if necessary
-    sig { params(indexable: IndexablePath).void }
-    def handle_change(indexable)
-      uri = indexable.to_uri.to_s
-      original_entries = @uris_to_entries[uri]
+    sig { params(uri: URI::Generic, indexable: IndexablePath).void }
+    def handle_change(uri, indexable)
```

Comment:
Are there cases where the `uri` would be different than `indexable.to_uri`? If there are, can we capture it in test cases?

Reply:
Also perhaps we can swap the arguments order, and have `uri` default to `indexable.uri`?

Reply:
Yeah, I think we need to change `IndexablePath` to be simply an `Indexable` and then include the URI there. I was just trying to avoid parsing the same URI twice, but the code is ugly.